product_cost_price_avco_sync/models/purchase_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'price_unit' in vals and not self.env.context.get(
+                'skip_cost_update'):
+            self.price_unit_update()
+        return res
+
+    def price_unit_update(self):
+        for line in self:
+            if line.state not in ['purchase', 'done']:
+                continue
+            line.move_ids.write({
+                'price_unit': line.price_unit,
```

Comment:
You must call here `_get_stock_move_price_unit` instead for taking into account several things like taxes price included, purchase discount module, etc.